controllers/kubeadmconfig_controller.go

```diff
@@ -259,24 +258,21 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 		return ctrl.Result{}, errors.New("Control plane already exists for the cluster, only KubeadmConfig objects with JoinConfiguration are allowed")
 	}
 
-	// Get certificates to improve security of discovery
-	certificates, err := r.getClusterCertificates(cluster)
-	if err != nil {
+	certificates := internalcluster.NewCertificates()
+	if err := certificates.GetCertificates(ctx, r.Client, cluster); err != nil {
 		log.Error(err, "unable to lookup cluster certificates")
 		return ctrl.Result{}, err
 	}
```

Comment:
Make sure we validate that all the certs are present and valid, otherwise it's an error

Reply:
This has been added as EnsureAllExist